browser/resources/settings/brave_sync_page/brave_sync_page.js

```diff
@@ -1,29 +1,98 @@
+    I18nBehavior,
+    WebUIListenerBehavior,
+  ],
 
-    properties: {},
+  properties: {
+    /**
+     * The current sync status, supplied by SyncBrowserProxy.
+     * @type {?settings.SyncStatus}
+     */
+    syncStatus_: Object,
+    syncLabel_: {
+      type: String,
+      computed: 'computeSyncLabel_(syncStatus_.firstSetupInProgress)'
+    },
+  },
 
-    /** @private {?settings.DefaultBraveSyncBrowserProxy} */
-    browserProxy_: null,
+  /** @private {?settings.SyncBrowserProxy} */
+  browserProxy_: null,
+  /** @private */
+  braveBrowserProxy_: null,
 
-    /** @override */
-    created: function() {
-      this.browserProxy_ = settings.DefaultBraveSyncBrowserProxyImpl.getInstance();
-    },
+  /** @override */
+  created: function() {
+    this.browserProxy_ = settings.SyncBrowserProxyImpl.getInstance();
+    this.braveBrowserProxy_ = settings.BraveSyncBrowserProxy.getInstance();
+  },
 
-    /** @override */
-    ready: function() {
-    },
-  });
-  })();
+  /** @private */
+  computeSyncLabel_() {
+    const isAlreadySetup = this.syncStatus_ !== undefined &&
+        !this.syncStatus_.firstSetupInProgress;
+    const key = isAlreadySetup ? 'braveSyncManageActionLabel' : 'braveSyncSetupActionLabel';
+    return I18nBehavior.i18n(key);
+  },
+
+  /** @override */
+  attached: function() {
+    const onSyncStatus = this.handleSyncStatus_.bind(this)
+    this.browserProxy_.getSyncStatus().then(onSyncStatus);
+    this.addWebUIListener('sync-status-changed',onSyncStatus);
+    this.addWebUIListener(
+        'sync-prefs-changed', this.handleSyncPrefsChanged_.bind(this));
+  },
+
+  /** @private */
+  onSyncTap_: function() {
+    // Users can go to sync subpage regardless of sync status.
+    const router = settings.Router.getInstance();
+    router.navigateTo(router.getRoutes().BRAVE_SYNC_SETUP);
+  },
+
+  /**
+   * Handler for when the sync state is pushed from the browser.
+   * @param {?settings.SyncStatus} syncStatus
+   * @private
+   */
+  handleSyncStatus_: function(syncStatus) {
+    console.debug('Sync Status Update', syncStatus)
+    this.syncStatus_ = syncStatus;
+  },
+
+  /**
+   * Handler for when the sync preferences are updated.
+   * @private
+   */
+  handleSyncPrefsChanged_: async function(syncPrefs) {
+    // Enforce encryption
+    if (this.syncStatus_ && !this.syncStatus_.firstSetupInProgress) {
+      if (!syncPrefs.encryptAllData) {
+        const syncCode = await this.braveBrowserProxy_.getSyncCode()
+        syncPrefs.encryptAllData = true;
+        syncPrefs.setNewPassphrase = true;
+        syncPrefs.passphrase = syncCode;
+        console.debug('sync set encryption', syncPrefs)
+        await this.browserProxy_.setSyncEncryption(syncPrefs)
+      } else if (syncPrefs.passphraseRequired) {
+        const syncCode = await this.braveBrowserProxy_.getSyncCode()
+        syncPrefs.setNewPassphrase = false;
+        syncPrefs.passphrase = syncCode;
+        console.debug('sync set encryption', syncPrefs)
```

Comment:
won't this leak the passphrase into the console (and potentially the log)?

Reply:
cc @petemill

Reply:
fixed in https://github.com/brave/brave-core/pull/5294/commits/247a37c0aca0207366827791456801efcd92cebe